File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string s_form = "";

        for(char &c : s){
            if(isalnum(c)){
                c = tolower(c);
                s_form+=c;
            }
        }


        int left = 0, right = s_form.length() - 1;

        while(left<right){
            if(s_form[right]!=s_form[left]) return false;
            left++;
            right--;
        }

        return true;

    }
};

File: 125-valid-palindrome/valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        s_lower = s.lower()
        s_f = ""
        for i in s_lower:
            if i.isalnum():
                s_f = s_f + i

        n = len(s_f)

        for i in range(0,n//2):
            if(s_f[i]!= s_f[n-i-1]):
                return False
        
        return True
        